pkg/virt-controller/services/renderresources.go

```diff
@@ -223,39 +207,47 @@ func WithAutoMemoryLimits(namespace string, namespaceStore cache.Store) Resource
 	}
 }
 
-func WithCPUPinning(cpu *v1.CPU, annotations map[string]string, additionalCPUs uint32) ResourceRendererOption {
+func WithCPUPinning(vmi *v1.VirtualMachineInstance, annotations map[string]string, additionalCPUs uint32) ResourceRendererOption {
 	return func(renderer *ResourceRenderer) {
+		cpu := vmi.Spec.Domain.CPU
 		vcpus := hardware.GetNumberOfVCPUs(cpu)
+		ioThreadCPUs := getIOThreadsCount(vmi)
 		if vcpus != 0 {
-			renderer.calculatedLimits[k8sv1.ResourceCPU] = *resource.NewQuantity(vcpus, resource.BinarySI)
+			totalCPUs := vcpus + ioThreadCPUs
+			renderer.vmLimits[k8sv1.ResourceCPU] = *resource.NewQuantity(totalCPUs, resource.BinarySI)
+			renderer.vmRequests[k8sv1.ResourceCPU] = *resource.NewQuantity(totalCPUs, resource.BinarySI) // Ensure requests match limits for dedicated CPUs
 		} else {
+			ioThreadsCount := resource.NewQuantity(ioThreadCPUs, resource.BinarySI)
```

Comment:
Per previous comment, this else statement can be entirely dropped.
The VMI mutator ensures that the vcpu is always different from 0.
We can do that in a follow up

Reply:
I would prefer if possible

Reply:
It works for me! Thanks